Utils/secrets.py

```diff
@@ -0,0 +1,109 @@
+""" fetch secrets by key from appropriate source according to environment """
+
+from enum import Enum
+import os
+
+
+TEAM_PREFIXES = {"HIVE", "BST"}
+PROJECT_IDS = {
+    "DEV": "amazing-city-422521-t1",
+    "ANA": "ana",
+    "BILL": "bill",
+    "STG": "stg",
+    "PROD": "prod",
+}
+
+
+class Env(Enum):
+    """ values are the key prefixes used in Secret Manager """
+    DEV = "DEV"
+    ANALYTICS = "ANA"
+    BILLING = "BILL"
+    STAGING = "STG"
+    PROD = "PROD"
+
+
+class SecretManager():
+    local_envs = {Env.DEV, Env.ANALYTICS}
+    project_ids = {env: PROJECT_IDS[env.value] for env in Env}
+
+    def __init__(self, env: str = None):
+        if env is None:
+            env = os.environ.get("ENV", "dev")
+        self.env = Env[env.upper()]
+        self.is_local_env = self.env in self.local_envs
+
+    def _make_full_key(self, key: str) -> str:
+        """
+        prepend the environment to the key
+
+        @param key: the name of the secret with format {team}_{secret_name}
+        @return: name of the secret in Secret Manager with format {env}_{team}_{secret_name}
```

Comment:
Note: this naming scheme is just my proposal to keep things organized and ownership clear. But if my assumption that there will always be different sets of secrets for the two teams doesn't hold, then let's discuss a better naming convention. E.g., if there will just be one prod service account instead of one for each team.